### PR TITLE
Add Schedule resource to Meteoroid

### PR DESCRIPTION
**Pull request by YujiAzama**
The following changes are included to add a schedule resource.

  - Schedules API
  - Schedule Model
  - Unit test code of schedule resource.
  - User document of schedule resource.

### CODE REVIEW

**Comment by ryo-ma** [comment]
Please confirm format by flake8.

File: meteoroid/core/libs/faas_driver.py [hunk 1]
```diff
@@ -240,3 +252,107 @@ def retrieve_result(self, result_id, fiware_service, fiware_service_path):
...
+            'name': data['name'],
+            'schedule': data['schedule'],
+            'function': function.id,
+            'startDate': invoke_data.get('startDate', ''),
+            'stopDate': invoke_data.get('stopDate', '')
+        }
+        return schedule
+
+    def retrieve_schedule(self, schedule, fiware_service, fiware_service_path):
+        OpenWhiskClient().retrieve_trigger(schedule.trigger_name, 'guest')
```
Comment (ryo-ma) [inline meteoroid/core/libs/faas_driver.py]: Do you need this line? I look like you are't using response.

File: meteoroid/core/serializers.py [hunk 2]
```diff
@@ -87,3 +88,19 @@ def __init__(self, *args, **kwargs):
...
+
+class ScheduleSerializer(serializers.ModelSerializer):
+    fiware_service = serializers.CharField(max_length=64, default='', allow_blank=True)
+    fiware_service_path = serializers.CharField(max_length=64, default='/')
+    trigger_name = serializers.CharField(max_length=64)
+    rule_name = serializers.CharField(max_length=64)
+    function = serializers.PrimaryKeyRelatedField(queryset=Function.objects.filter())
+
+    def __init__(self, *args, **kwargs):
+        self.faas_schedule_data = kwargs.pop('faas_schedule_data', '')
```
Comment (ryo-ma) [inline meteoroid/core/serializers.py]: Are you using the **self.faas_schedule_data**?

File: meteoroid/core/libs/faas_driver.py (hunk 1 above)
Comment (YujiAzama) [reply meteoroid/core/libs/faas_driver.py]: This line has been deleted.

File: meteoroid/core/serializers.py (hunk 2 above)
Comment (YujiAzama) [reply meteoroid/core/serializers.py]: self.faas_schedule_data has been deleted.

**Comment by YujiAzama** [comment]
I updated and fixed conflicts.

File: meteoroid/core/models.py [hunk 3]
```diff
@@ -59,3 +59,13 @@ def __str__(self):
...
+
+
+class Schedule(FIWAREBase):
+    trigger_name = models.CharField(max_length=64)
+    rule_name = models.CharField(max_length=64)
+    name = models.CharField(max_length=64)
+    function = models.ForeignKey(Function, on_delete=models.CASCADE, related_name='functions')
+
+    class Meta:
+        unique_together = ('fiware_service', 'fiware_service_path')
```
Comment (ryo-ma) [inline meteoroid/core/models.py]: I will get the following Error if I register a different schedule twice. The fiware_service and fiware_service_path of the two schedulers are the same and have different names.
Please make sure unique constraint.
`AssertionError: You cannot call .save() on a serializer with invalid data.`

Comment (YujiAzama) [reply meteoroid/core/models.py]: Thank you for trying creating multiple schedules resource.
I added name key to unique_together.

**Review by ryo-ma — APPROVED** [review]

**Comment by ryo-ma** [comment]
I approved.
@hhayashiool Is there any problem?

**Pull request merged**